src/BalanceTx.purs

```diff
@@ -0,0 +1,831 @@
+  show = show <<< flattenValue <<< unwrap
+
+newtype BalanceNonAdaOutsFailure = BalanceNonAdaOutsFailure BalanceNonAdaOutsFailureReason
+
+derive instance newtypeBalanceNonAdaOutsFailure :: Newtype BalanceNonAdaOutsFailure _
+
+derive newtype instance showBalanceNonAdaOutsFailure :: Show BalanceNonAdaOutsFailure
+
+data BalanceNonAdaOutsFailureReason = InputsCannotBalanceNonAdaTokens
+
+instance showBalanceNonAdaOutsFailureReason :: Show BalanceNonAdaOutsFailureReason where
+  show InputsCannotBalanceNonAdaTokens = "Not enough inputs to balance tokens."
+
+-- newtype AddSignatoriesFailure = AddSignatoriesFailure AddSignatoriesFailureReason
+
+-- derive instance newtypeAddSignatoriesFailure :: Newtype AddSignatoriesFailure _
+-- derive newtype instance showAddSignatoriesFailure :: Show AddSignatoriesFailure
+
+-- data AddSignatoriesFailureReason = SigningAddressNotFound Address
+
+-- instance showAddSignatoriesFailureReason :: Show AddSignatoriesFailureReason where
+--   show (SigningAddressNotFound address) =
+--     "Signing address not found for: " <> show address
+
+-- | Represents that an error reason should be impossible
+data Impossible = Impossible
+
+newtype BuildTxRawFailure = BuildTxRawFailure BuildTxRawFailureReason
+
+derive instance newtypeBuildTxRawFailure :: Newtype BuildTxRawFailure _
+derive newtype instance showBuildTxRawFailure :: Show BuildTxRawFailure
+
+data BuildTxRawFailureReason = CannotBuildTxRaw
+
+instance showBuildTxRawFailureReason :: Show BuildTxRawFailureReason where
+  show CannotBuildTxRaw = "Cannot build raw transaction."
+
+newtype CalculateMinFeeFailure = CalculateMinFeeFailure CalculateMinFeeFailureReason
+
+derive instance newtypeCalculateMinFeeFailure :: Newtype CalculateMinFeeFailure _
+
+derive newtype instance showCalculateMinFeeFailure :: Show CalculateMinFeeFailure
+
+data CalculateMinFeeFailureReason = CannotCalculateMinFee
+
+instance showCalculateMinFeeFailureReason :: Show CalculateMinFeeFailureReason where
+  show CannotCalculateMinFee = "Cannot calculate minimum fee."
+
+--------------------------------------------------------------------------------
+-- Type aliases, temporary placeholder types and functions
+--------------------------------------------------------------------------------
+-- Output utxos with the amount of lovelaces required.
+type MinUtxos = Array (TransactionOutput /\ BigInt)
+
+-- TO DO: convert utxosAt from Ogmios to Transaction space.
+utxosAt :: Address -> QueryM UtxoM
+utxosAt = undefined
+
+-- Taken from https://playground.plutus.iohkdev.io/doc/haddock/plutus-ledger-constraints/html/Ledger-Constraints-OffChain.html#t:UnbalancedTx
+-- to demonstrate what is required. This is temporary to make the compiler happy.
+-- I haven't copied it exactly, instead just a minimal version given our
+-- incomplete Transaction.purs types. Also requiredSignatures is already in
+-- TxBody. I've ignored validity interval too.
+newtype UnbalancedTransaction = UnbalancedTransaction
```

Comment:
Vladimir's implementation was merged, right?

Reply:
I was going to leave this for a second PR, just to reduce the amount of potentially breaking dependencies if it changes again.